cpp/util/profiler.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2015-2015 the openage authors. See copying.md for legal info.
+
+#ifndef OPENAGE_UTIL_PROFILER_H_
+#define OPENAGE_UTIL_PROFILER_H_
+
+#include <map>
+#include <vector>
+#include <chrono>
+#include <memory>
+#include <list>
+
+#define MAX_DURATION_HISTORY 100
+
+namespace openage {
+namespace util {
+
+struct CategorieTimeData {
+	std::chrono::high_resolution_clock::time_point start;
+	std::chrono::high_resolution_clock::duration duration;
+	std::list<std::chrono::high_resolution_clock::duration> history;
```

Comment:
Isn't `std::list` rather memory-heavy? Not Landau Notation-wise, but I imagine that it involves a lot of slow small heap allocations 'n stuff.
`std::vector` would be preferable if heap allocation is required (you'd have to manually manage the cyclic access thingie, but that's just one extra variable and a few lines of code).

However, since we know the history size as a compile-time constant, you can simply use `std::array` instead of `std::vector`, which uses no heap allocation at all.